tinygrad/codegen/kernel.py

```diff
@@ -529,6 +544,27 @@ def get_tc_swizzle_st(shape, local_perm, upcast_perm):
     del fixup_ast
     return graph_rewrite(fixed_ast, view_left, name="fixup optimized AST")
 
+  def apply_lds(self, ast) -> UOp:
+    def transform(ctx:tuple[Kernel, set[UOp]], global_access:UOp):
+      if (buf := global_access.src[0]).op is not Ops.DEFINE_GLOBAL or buf.arg in ctx[1] or not (k := ctx[0]).lds[buf.arg]: return None
+      ctx[1].add(buf.arg)
+      global_st: ShapeTracker = global_access.src[1].arg
+      gd, fr, fu, shape = k.global_dims, k.first_reduce, k.first_upcast, []
```

Comment:
Why!?! Just write them out each time so it's readable. Don't prematurely optimize.